shark_turbine/kernel/wave/codegen.py

```diff
@@ -247,13 +258,9 @@ def _process_mul_add_ops(term, is_mul):
                     numerator = arith_d.SubIOp(zero, numerator)
                 mul = lambda x: x
                 if abs(term.p) != 1:
-                    mul = lambda x: arith_d.MulIOp(x, numerator)
-                add = lambda x: arith_d.AddIOp(
-                    arith_d.MulIOp(x, denominator), numerator
-                )
-                operation = lambda x, is_mul: arith_d.DivSIOp(
-                    mul(x) if is_mul else add(x), denominator
-                )
+                    mul = _get_mul(numerator)
```

Comment:
I don't see what the problem was. Were the values of numerator and denominator being captured by the lambda and so not changing?

Reply:
Ah I see, that makes sense. Since in python assignment is essentially just "labeling", the lambda captures the label and not the object being pointed to.